File: 프로그래머스/unrated/181925. 수 조작하기 2/수 조작하기 2.java
class Solution {
    public String solution(int[] numLog) {
        String answer = "";

        for (int i = 0; i < numLog.length - 1; i++) {
            int a = numLog[i];
            int b = numLog[i + 1];

            if (Math.abs(a - b) == 1) {
                if (b > a) {
                    answer += "w";
                } else {
                    answer += "s";
                }
            } else {
                if (b > a) {
                    answer += "d";
                } else {
                    answer += "a";
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/unrated/181925. 수 조작하기 2/수 조작하기 2.py
def solution(numLog):
    answer = ''
    
    for i in range(len(numLog)-1):
        s = numLog[i] - numLog[i+1]
        if s == 1:
            answer += 's'
        elif s == -1:
            answer+='w'
        elif s == 10:
            answer+='a'
        else:
            answer+='d'
        
    return answer